### PR TITLE
chore(entities): consolidate entity property APIs

### LINKED ISSUE #11427: fetching entities with boolean values in metadata/privatesettings fails
somewhere https://github.com/Elgg/Elgg/blob/master/views/default/admin/upgrades.php#L13 is cast to an `int` and therefore the query fails. If I change `false` to `0` or `'0'` it works

@hypeJunction help please

### PR DESCRIPTION
**Pull request by hypeJunction**
Aligns private settings table logic with metadata and annotations.
Cleans up remaining redunant API around entity properties.
Adds more tests.
Fixes issues with boolean casting in ege* by metadata, annotations
and private settings.

Fixes #11427

### CODE REVIEW

File: engine/lib/deprecated-3.0.php [hunk 1]
```diff
@@ -1079,4 +1079,177 @@ function sanitise_filepath($path, $append_slash = true) {
...
+ * @param int    $entity_guid The entity GUID
+ * @param string $name        The name of the setting
+ *
+ * @return mixed The setting value, or null if does not exist
+ * @see set_private_setting()
+ * @see get_all_private_settings()
+ * @see remove_private_setting()
+ * @see remove_all_private_settings()
+ */
+function get_private_setting($entity_guid, $name) {
```
Comment (hypeJunction) [inline engine/lib/deprecated-3.0.php]: These need to set ignored access to stay BC

**Review by mrclay — APPROVED** [review]

**Comment by jdalsem** [comment]
i think this is good for Elgg, but it breaks so much... data storage of booleans/ int like strings.... retrival via elgg_get_entities... this PR requires a better mention in the upgrade notes...

**Comment by hypeJunction** [comment]
Does it though?

**Comment by jdalsem** [comment]
booleans now get cast to ints... it was strings... major difference... false being cast to 0 in ege is new for private settings

this test for example fails before this PR https://github.com/Elgg/Elgg/pull/11543/files#diff-8e7a21e63a7e0490dcb731018e9f0806R156

**Comment by hypeJunction** [comment]
So, what do you want here? I hardly ever use private settings to would have noticed any changes

**Comment by jdalsem** [comment]
i do not know... do we want to break this? i do not know how to deal with the data migration then...

**Comment by hypeJunction** [comment]
Well, there is no way to migrate data. We can't second guess if an empty string is an empty string or a false.

**Comment by jdalsem** [comment]
indeed... do we accept that unmigratable data?

**Comment by hypeJunction** [comment]
I'd say mention in upgrade docs, and tell plugins to migrate their own data. They know better what the expected values are.

**Review by jeabakker — APPROVED** [review]

**Comment by jeabakker** [comment]
accept potential problems in plugins. If we find problems in core will fix them as needed

**Pull request merged**